client/components/command-palette/use-command-palette.tsx

```diff
@@ -88,13 +88,53 @@ export const useCommandPalette = ( {
 	const { currentSiteId } = useCurrentSiteRankTop();
 
 	// Call the generateCommandsArray function to get the commands array
-	let commands = useCommandsArrayWpcom( { setSelectedCommandName } );
+	const commands = useCommandsArrayWpcom( {
+		setSelectedCommandName,
+	} );
 
-	if ( 'function' === typeof filter ) {
-		commands = commands.filter( filter );
-	}
+	const currentPath = useSelector( ( state: object ) => getCurrentRoute( state ) );
+
+	// Filter out commands that have context
+	const commandHasContext = ( paths: string[] = [] ): boolean =>
+		paths.some( ( path ) => currentPath.includes( path ) ) ?? false;
+
+	// Ensure that View My Sites command ranks the highest
+	const viewMySitesSort = ( a: { name: string }, b: { name: string } ) => {
+		const isViewMySitesWithContextual = a.name === 'viewMySites';
+		const isViewMySitesNoContextual = b.name === 'viewMySites';
+
+		if ( isViewMySitesWithContextual && ! isViewMySitesNoContextual ) {
+			return -1; // "viewMySites" comes first over contextual commands
+		} else if ( ! isViewMySitesWithContextual && isViewMySitesNoContextual ) {
+			return 1; // "viewMySites" comes first over regular commands
+		}
+
+		return 0; // no change in order
+	};
+
+	// Filter commands to exclude "viewMySites" when the current path is /sites
+	const filteredCommands = commands.filter(
+		( command ) => ! ( command.name === 'viewMySites' && currentPath === '/sites' )
+	);
+
+	// Sort the filtered commands with the contextual commands ranking higher than general in a given context
+	const sortedCommands = filteredCommands.sort( ( a, b ) => {
+		const hasContextCommand = commandHasContext( a.context );
+		const hasNoContext = commandHasContext( b.context );
+
+		if ( hasContextCommand && ! hasNoContext ) {
+			return -1; // commands with context come first if there is a context match
+		} else if ( ! hasContextCommand && hasNoContext ) {
+			return 1; // commands without context set
+		}
+
+		return 0; // no change in order
+	} );
+
+	// Combine the sorted "viewMySites" commands and the main sorted commands
+	const finalSortedCommands = [ ...sortedCommands ].sort( viewMySitesSort );
```

Comment:
...after the commands are sorted.
1. Let's remove the viewMySites command and store it in a variable.
2. If we are on `/sites` that's it.
3. If we are on a different page, let's prepend that command that we removed. `[viewMySitesCommand, ...sortedCommands]`

Optionally, we could encapsulate that logic on a different hook.

Reply:
I made the updates based on the suggestion! I like this approach - I find the logic much simpler ❤️